src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -665,7 +751,39 @@ public static String Join<T>(String separator, IEnumerable<T> values)
             }
         }
 
-        // Joins an array of strings together as one string with a separator between each original string.
+        // Joins a string IEnumerable together as one string with a char separator between each original string.
+        //
+        [ComVisible(false)]
+        public static String Join(Char separator, IEnumerable<String> values) {
```

Comment:
Do we need this overload?  The use case is covered entirely by the ```Join(char, IEnumerable<T>)``` overload, and the only benefit I see to this one is avoiding per-element a branch and a ToString call (which will just return ```this```).  Is the performance difference between, say, ```string.Join('a', Enumerable.Repeat("b", 10000000))``` and  ```string.Join<string>('a', Enumerable.Repeat("b", 10000000))``` different enough that it's worth having the extra overload?